src/modules/settings/IdentityProvidersTab.tsx

```diff
@@ -0,0 +1,343 @@
+          >
+            <Trash2 size={14} className="mr-2" />
+            Delete
+          </DropdownMenuItem>
+        </DropdownMenuContent>
+      </DropdownMenu>
+    </div>
+  );
+}
+
+export default function IdentityProvidersTab() {
+  const { permission } = usePermissions();
+  const { mutate } = useSWRConfig();
+  const { data: providers, isLoading } =
+    useFetchApi<IdentityProvider[]>("/identity-providers");
+
+  const [modalOpen, setModalOpen] = useState(false);
+  const [editProvider, setEditProvider] = useState<IdentityProvider | null>(
+    null,
+  );
+
+  const [sorting, setSorting] = useLocalStorage<SortingState>(
+    "netbird-table-sort-identity-providers",
+    [
+      {
+        id: "name",
+        desc: false,
+      },
+    ],
+  );
+
+  const handleEdit = (provider: IdentityProvider) => {
+    setEditProvider(provider);
+    setModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+    setEditProvider(null);
+  };
+
+  const columns: ColumnDef<IdentityProvider>[] = [
+    {
+      accessorKey: "name",
+      header: ({ column }) => (
+        <DataTableHeader column={column}>Name</DataTableHeader>
+      ),
+      sortingFn: "text",
+      cell: ({ row }) => (
+        <div className="flex items-center gap-3">
+          {idpIcons[row.original.type] || <KeyRound size={16} className="text-nb-gray-400" />}
+          <span className="font-medium">{row.original.name}</span>
+        </div>
+      ),
+    },
+    {
+      accessorKey: "type",
+      header: ({ column }) => (
+        <DataTableHeader column={column}>Type</DataTableHeader>
+      ),
+      cell: ({ row }) => (
+        <span className="text-nb-gray-400">
+          {idpTypeLabels[row.original.type] || row.original.type}
+        </span>
+      ),
+    },
+    {
+      id: "actions",
+      accessorKey: "id",
+      header: "",
+      cell: ({ row }) => (
+        <ActionCell provider={row.original} onEdit={handleEdit} />
+      ),
+    },
+  ];
+
+  return (
+    <Tabs.Content value={"identity-providers"} className={"w-full"}>
+      <div className={"p-default py-6"}>
+        <Breadcrumbs>
+          <Breadcrumbs.Item
+            href={"/settings"}
+            label={"Settings"}
+            icon={<SettingsIcon size={13} />}
+          />
+          <Breadcrumbs.Item
+            href={"/settings?tab=identity-providers"}
+            label={"Identity Providers"}
+            icon={<FingerprintIcon size={14} />}
+            active
+          />
+        </Breadcrumbs>
+        <div className={"flex items-start justify-between"}>
+          <div>
+            <h1>Identity Providers</h1>
+            <Paragraph>
+              Configure identity providers for user authentication in your
+              network.
+            </Paragraph>
+          </div>
+        </div>
+      </div>
+
+      <IdentityProviderModal
+        open={modalOpen}
+        onClose={handleCloseModal}
+        provider={editProvider}
+      />
```

Comment:
```suggestion
      <IdentityProviderModal
        open={modalOpen}
        key={modalOpen ? 1 : 0}
        onClose={handleCloseModal}
        provider={editProvider}
      />
```
This ensures that the component re-renders whenever the modal closes/opens to set the initial state (e.g., type, client_id, etc. when a provider is passed)